sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessagesView.java

```diff
@@ -52,19 +78,158 @@ public LabMessagesView() {
-				listComponent.getGrid().reload();
-			}
+			fetchLabmessages();
 		}, ValoTheme.BUTTON_PRIMARY));
+
+		if (isBulkEditAllowed()) {
+			btnEnterBulkEditMode = ButtonHelper.createIconButton(Captions.actionEnterBulkEditMode, VaadinIcons.CHECK_SQUARE_O, e -> {
+				enterBulkEditMode();
+			}, ValoTheme.BUTTON_PRIMARY);
+			btnEnterBulkEditMode.setVisible(!viewConfiguration.isInEagerMode());
+			addHeaderComponent(btnEnterBulkEditMode);
+
+			btnLeaveBulkEditMode = ButtonHelper.createIconButton(Captions.actionLeaveBulkEditMode, VaadinIcons.CLOSE, e -> {
+				leaveBulkEditMode();
+			}, ValoTheme.BUTTON_PRIMARY);
+			btnLeaveBulkEditMode.setVisible(viewConfiguration.isInEagerMode());
+			addHeaderComponent(btnLeaveBulkEditMode);
+		}
+
+		VerticalLayout gridLayout = new VerticalLayout();
+		addComponent(gridLayout);
+
+		gridLayout.addComponent(createStatusFilterBar());
+
+		grid = new LabMessageGrid(criteria);
+		gridLayout.addComponent(grid);
+		grid.getDataProvider().addDataProviderListener(e -> updateStatusButtons());
+
+		gridLayout.setMargin(true);
+		styleGridLayout(gridLayout);
 	}
 
 	@Override
 	public void enter(ViewChangeListener.ViewChangeEvent event) {
-		listComponent.reload(event);
+		String params = event.getParameters().trim();
+		if (params.startsWith("?")) {
+			params = params.substring(1);
+			criteria.fromUrlParams(params);
+		}
+		updateStatusButtons();
+		grid.reload();
+	}
+
+	public HorizontalLayout createStatusFilterBar() {
+		HorizontalLayout statusFilterLayout = new HorizontalLayout();
+		statusFilterLayout.setWidthFull();
+		statusFilterLayout.setMargin(false);
+		statusFilterLayout.setSpacing(true);
+		statusFilterLayout.addStyleName(CssStyles.VSPACE_3);
+
+		statusButtons = new HashMap<>();
+
+		Button allButton = createProcessedFilterButton(Captions.all, null);
+		statusFilterLayout.addComponent(allButton);
+		activeStatusButton = allButton;
+
+		statusFilterLayout.addComponent(createProcessedFilterButton(Captions.labMessageUnprocessed, Boolean.FALSE));
+		statusFilterLayout.addComponent(createProcessedFilterButton(Captions.labMessageProcessed, Boolean.TRUE));
+
+		HorizontalLayout actionButtonsLayout = new HorizontalLayout();
+		actionButtonsLayout.setSpacing(true);
+		statusFilterLayout.addComponent(actionButtonsLayout);
+		statusFilterLayout.setComponentAlignment(actionButtonsLayout, Alignment.TOP_RIGHT);
+		statusFilterLayout.setExpandRatio(actionButtonsLayout, 1);
+
+		bulkOperationsDropdown = createBulkOperationsDropdown();
+		actionButtonsLayout.addComponent(bulkOperationsDropdown);
+
+		return statusFilterLayout;
+	}
+
+	private MenuBar createBulkOperationsDropdown() {
+		final List<MenuBarHelper.MenuBarItem> menuBarItems = new ArrayList<>();
+
+		menuBarItems.add(new MenuBarHelper.MenuBarItem(I18nProperties.getCaption(Captions.bulkDelete), VaadinIcons.TRASH, mi -> {
+			ControllerProvider.getLabMessageController().deleteAllSelectedItems(grid.asMultiSelect().getSelectedItems(), () -> navigateTo(criteria));
```

Comment:
Here (or in the methods called here), it would be good to check whether a labMessage was processed in the meantime (between the time a deletion was considered possible and the actual deletion). Might be somehing that has to go into the LabMessageService.delete(...) method.

Reply:
I can still delete lab messages that were processed in the meantime via bulk edit mode.